### PR TITLE
Makes tar-fs able to pack files that are actively being written to.

### PR DESCRIPTION
**Pull request by asyed94**
Pretty simple change that allows tar-fs to be robust in more conditions, while maintaining its expected behavior.

### CODE REVIEW

**Comment by mafintosh** [comment]
Can you explain a bit more what this fixes?

File: index.js [hunk 1]
```diff
@@ -138,7 +138,7 @@ exports.pack = function (cwd, opts) {
     var entry = pack.entry(header, onnextentry)
     if (!entry) return
 
-    var rs = mapStream(xfs.createReadStream(path.join(cwd, filename)), header)
+    var rs = mapStream(xfs.createReadStream(path.join(cwd, filename), { start: 0, end: header.size > 0 ? header.size - 1 : header.size }), header)
```
Comment (asyed94) [inline index.js:141]: Ah, I remember what this does now (it's been a while).

`tar.pack('some_dir')` works by getting a file from `some_dir`, creating a ReadStream for that file, then piping the ReadStream to a `tar` entry, which triggers this whole process for the next file in `some_dir` upon completion of the pipe, until all files from `some_dir` are packed.

However, if a file in `some_dir` is continuously being written to (e.g. a log file), that file has no EOF, and piping it's ReadStream to an entry will never complete, holding up the whole packing process and causing the tar file for `some_dir` to never be made.

This change adds `start` and `end` parameters to the ReadStream of a file based on the files size at that time. Adding these parameters ensures that piping the ReadStream to an entry will reach completion even if the file has no EOF, allowing the tar packing process to continue.

Comment (mafintosh) [reply index.js:141]: Makes sense

**Comment by mafintosh** [comment]
Thanks for explaining this. If you can remove the lock file then this is good to go! :)

**Comment by asyed94** [comment]
Went ahead and removed the package-lock.json. I think that got in by accident :sweat_smile:

**Comment by mafintosh** [comment]
2.1.0

**Comment by mafintosh** [comment]
Thanks for the contribution! Appreciate it.

**Pull request merged**